cpp/box/export-box.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2010-2023 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#ifndef EXPORT_BOX_H
+#define EXPORT_BOX_H
+
+#include "Box.h"
+
+#include <nanobind/ndarray.h>
+namespace nb = nanobind;
+
+namespace freud { namespace box {
+
+template<typename T, typename shape = nb::shape<-1, 3>>
+using nb_array = nb::ndarray<T, shape, nb::device::cpu, nb::c_contig>;
+
+void makeAbsolutePython(Box box, nb_array<float, nb::shape<-1, 3>> vecs,
```

Comment:
Avoid making extra copies of the box.
```suggestion
void makeAbsolutePython(const Box& box, nb_array<float, nb::shape<-1, 3>> vecs,
```
Apply this similarly to all methods.

Is the intention to pass `Box` around by copies instead of reference counting? This is OK as `Box` is a lightweight object. Larger classes should definitely be wrapped in `std::shared_ptr` to share reference counting with `nanobind`.

Reply:
I didn't think of this originally, you are right that we shouldn't make unnecessary copies of the box class. After reading the [documentation](https://nanobind.readthedocs.io/en/latest/porting.html#shared-pointers-and-holders), I think the `std::shared_ptr` approach will work best.